components/ledger/pkg/storage/sqlstorage/ledger/store.go

```diff
@@ -106,10 +106,7 @@ func (s *Store) RunInTransaction(ctx context.Context, f func(ctx context.Context
 	newStore.isInitialized = s.isInitialized
 
 	defer func() {
-		// TODO(gfyrag): add recover() to rollback txs
-		if err != nil {
-			_ = tx.Rollback()
-		}
+		_ = tx.Rollback()
```

Comment:
We rollback everytime ?

Reply:
Why removing the error check ?

Reply:
The error check was not ok.
If the 'tx.Commit()' return an error, the rollback is not called.
The error check is not needed, as calling tx.Rollback() will have no effect if the transaction is already committed.